TombEngine/Game/effects/smoke.cpp

```diff
@@ -191,6 +191,51 @@ namespace TEN::Effects::Smoke
 		s.room = LaraItem->RoomNumber;
 	}
 
+	void TriggerGunSmokeParticles(ItemInfo* item, int x, int y, int z, int xv, int yv, int zv, byte initial, byte count, short roomNumber)
```

Comment:
Can we make another overload of `TriggerGunSmokeParticles` utilize this variation under the hood to avoid copypaste?

Reply:
Wanted to have something fast to test, i think @Sezzary will be able to change it correctly and @Nemoel-Tomo to upgrade it to look nice ingame.

Reply:
Done, just added roomNumber as a new argument.